StringHelper.cpp

```diff
@@ -18,21 +18,19 @@ std::string StringHelper::ConvertToUpper(const std::string& str)
 }
 
 // Returns a new vector with matching strings removed. Case insensitive.
-std::vector<std::string> StringHelper::RemoveStrings(const std::vector<std::string>& stringsToSearch, const std::vector<std::string>& stringsToRemove)
+std::vector<std::string> StringHelper::RemoveStrings(std::vector<std::string> stringsToSearch, const std::vector<std::string>& stringsForRemoval)
```

Comment:
These changes look good. I had to think about this one a bit. I noticed there is now an implicit vector copy done in the parameter list while calling the function, rather than an explicit one within the function. That nicely cuts down on code. The loop counting changed, looks like you were fixing a bug where the old code started at the end of the vector and used `++i`, while the new code counts down according to the recently discussed idiom.

I believe this code avoids iterator invalidation problems after calling [`erase`](https://en.cppreference.com/w/cpp/container/vector/erase). After `erase`, any iterators at or beyond the point of erasure are invalidated.

I wonder if this code should make use of [`remove_if`](https://en.cppreference.com/w/cpp/algorithm/remove). It would probably result in less copying in the case where multiple values are removed. That might be a subject for another branch though.

Reply:
Yeah, the ++ in the loop was the bug allowing for only sing file removal from archives. I had only been testing removal of single files in the past. When reviewing the code I noticed it looked odd to be a ++ and tested better.

Generally agree this passage is difficult to read and could be improved on.